src/ascent/runtimes/ascent_transmogrifier.cpp

```diff
@@ -107,6 +109,151 @@ conduit::Node* Transmogrifier::low_order(conduit::Node &dataset)
 #endif
 }
 
+bool Transmogrifier::is_poly(const conduit::Node &doms)
+{
+  const int num_domains = doms.number_of_children();
+
+  for (int i = 0; i < num_domains; i ++)
+  {
+    const conduit::Node &dom = doms.child(i);
+    conduit::NodeConstIterator itr = dom["topologies"].children();
+    while (itr.has_next())
+    {
+      const conduit::Node &topo = itr.next();
+      if (topo.has_child("elements"))
+      {
+        if (topo["elements"].has_child("shape"))
+        {
+          if (topo["elements/shape"].as_string() == "polyhedral" || 
+              topo["elements/shape"].as_string() == "polygonal")
+          {
+            return true;
+          }
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+// to_poly assumes that the node n is polyhedral
+void Transmogrifier::to_poly(conduit::Node &doms, conduit::Node &to_vtkh)
+{
+  const int num_domains = doms.number_of_children();
+
+  for (int i = 0; i < num_domains; i ++)
+  {
+    const conduit::Node &dom = doms.child(i);
+    std::vector<std::string> poly_topos;
+    conduit::Node &res = to_vtkh.append();
+
+    // we know it must have a child "topologies" b/c otherwise it is not valid blueprint
+    conduit::NodeConstIterator itr = dom["topologies"].children();
+    while (itr.has_next())
+    {
+      const conduit::Node &topo = itr.next();
+      if (topo.has_child("elements"))
+      {
+        if (topo["elements"].has_child("shape"))
+        {
+          if (topo["elements/shape"].as_string() == "polyhedral" || 
+              topo["elements/shape"].as_string() == "polygonal")
+          {
+            poly_topos.push_back(topo.name());
+          }
+        }
+      }
+    }
+
+    res.set_external(dom);
```

Comment:
All the tests work without me having to delete the parts of the nodes that are being overwritten; should I not worry about it then?

Reply:
it should magically work :-)